geoalchemy2/elements.py

```diff
@@ -131,11 +132,41 @@ class WKBElement(_SpatialElement):
 
     Note: you can create ``WKBElement`` objects from Shapely geometries
     using the :func:`geoalchemy2.shape.from_shape` function.
+
+
+    WKB struct {
+        byte    byteOrder;
+        uint32  wkbType;
+        uint32  SRID;
+        struct  geometry;
+    }
+
+    byteOrder enum {
+        WKB_XDR = 0,  // Most Significant Byte First
+        WKB_NDR = 1,  // Least Significant Byte First
+    }
+
     """
 
     geom_from = 'ST_GeomFromWKB'
     geom_from_extended_version = 'ST_GeomFromEWKB'
 
+    def __init__(self, data, srid=-1, extended=False):
+        if extended and srid == -1:  # unpack srid from WKB struct
```

Comment:
I was wondering if the `srid == -1` statement is relevent. Maybe we could remove it and raise an error if the given srid is not `-1` is not consistent with the one contained in the data?

Reply:
I was not sure either. I also like the idea of avoiding the data unpacking when the srid is specified in the Geometry type.

For example, if I am always transforming to, let's say, EPSG:3857, I can write my own function class like this:

```py
class ST_Transform3857(GenericFunction):
    type = Geometry(srid=3857)
    name = 'ST_Transform'
    identifier = 'ST_Transform3857'
```

Or I can set the type at query time:

```py
s = select([func.ST_Transform(t.c.geom, 4326), 3857, _type=Geometry(srid=3857))])
```

So, with the way @SergeBouchut wrote the code we avoid the data unpacking in the above two cases.

Reply:
Ok I see. It is clearly more efficient in those cases but the user is responsible to ensure that the 2 srid are consistent. Maybe we could just add a word in the doc to make it clear?

Reply:
Agreed.